althea_kernel_interface/src/interface_tools.rs

```diff
@@ -24,6 +25,15 @@ impl KernelInterface {
         self.run_command("ip", &["link", "del", "dev", name])?;
         Ok(())
     }
+
+    pub fn iface_status(&self, iface: &str) -> Result<String, Error> {
+        // cat so we can mock
+        let output = self.run_command("cat", &[&format!("/sys/class/net/{}/operstate", iface)])?;
```

Comment:
so we cat here rather than just reading the file for easy mocking? Could we mock the file stream instead? Not sure if that's actually better though.

Reply:
We already have the infra to mock commands, so this is the past of least resistance (we almost ever read files)